components/CallCenterPicker.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
+import generateSaml from "../pages/api/generateSaml";
+import { SsoDetails } from "../pages/api/types";
+import styles from "../styles/Home.module.css";
+
+interface Props {
+  cognitoGroups: string[];
+  idToken: string;
+  pickerWasPressed: boolean;
+  setPickerWasPressed: React.Dispatch<React.SetStateAction<boolean>>;
+  setSsoDetails: React.Dispatch<React.SetStateAction<SsoDetails | undefined>>;
+  setError: React.Dispatch<React.SetStateAction<string | undefined>>;
+}
+
+/**
+ * Pick one call center with a radio button type of form when a user is
+ * configured with multiple call centers.
+ *
+ * @param cognitoGroups The groups (call centers) this user belongs to
+ * @param idToken User's Cognito token necessary for calling `generateSaml`
+ * @param pickerWasPressed Boolean state variable whether this picker has been
+ * clicked/submitted
+ * @param setPickerWasPressed Setter for the above state variable
```

Comment:
I think this will be a matter of opinion, but the use of `pickerWasPressed` is a bit confusing to me-- it looks like both as a prop and as state within this component. Maybe `CallCenterPicker` could feel more independent if it's named `onPress` while in this component?

Reply:
Maybe `hasBeenClicked` or something?